### PR TITLE
show all existing PVCs as dropdown under volumes for a new notebook server

### PR DESCRIPTION
**Pull request by lalithvaka**
Resolves issue #4883 
This PR enhances two things
1. Shows all Existing PVCs in the profile as a dropdown while creating a new notebook server under both workspace / data volumes when type drop down selected as "Existing"
2. Hide Mode and Size form fields under volumes when selected existing PVCs so that its not confusing

Follow are the screenshots of before and after
**Before the fix**
image

**After the fix**
image

### CODE REVIEW

**Comment by lalithvaka** [comment]
@avdaredevil , please let me know if you can review this PR as well?

File: components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.html [hunk 1]
```diff
@@ -1,42 +1,49 @@
...
+        <mat-option *ngFor="let i of types"
         [disabled]="newTypeIsDisabled()"
         [matTooltip]="newTypeTooltip()"
-        value="New"
-        >New</mat-option
-      >
-      <mat-option value="Existing">Existing</mat-option>
-    </mat-select>
-  </mat-form-field>
+        value={{i}}>{{i}}</mat-option>
```
Comment (avdaredevil) [inline components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.html:10]:
```suggestion
        value="{{i}}">{{i}}</mat-option>
```

File: components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.ts [hunk 2]
```diff
@@ -58,6 +63,14 @@ export class VolumeComponent implements OnInit, OnDestroy {
     }
   }
 
+// ----- onChange of the New / Existing Volume -----
+  selectType(event): void {
+    this.typeSelected = event.value;
+    if (this.typeSelected === 'New') {
+      this.volume.controls.name.setValue(this.currentVolName);
+    }
```
Comment (avdaredevil) [inline components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.ts:71]: *nit*: Prevent unnecessary code depth
```suggestion
    if (this.typeSelected !== 'New') return;
    this.volume.controls.name.setValue(this.currentVolName);
```

File: components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.html (hunk 1 above)
Comment (lalithvaka) [reply components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.html:10]: Will test this and commit again - Definitely not an expert on right way to code this.

File: components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.ts (hunk 2 above)
Comment (lalithvaka) [reply components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.ts:71]: Will test this and commit again

File: components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.html (hunk 1 above)
Comment (lalithvaka) [reply components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.html]: Submitted the commit again after testing.

File: components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.ts (hunk 2 above)
Comment (lalithvaka) [reply components/jupyter-web-app/frontend/src/app/resource-form/volume/volume.component.ts]: Updated and submitted the commit

**Comment by kimwnasptd** [comment]
@avdaredevil thanks for taking a look!

@lalithvaka I'll take a more thorough look at this asap.
With a first glance could you make the `Mount Point` input wide enough to fill the space left from the two previous inputs, when selecting an existing volume?

**Comment by lalithvaka** [comment]
@kimwnasptd Can you please review this? If you have any additional changes, I will submit the pull request. Am wondering if this can be pushed to 1.1 release as well!

**Comment by lalithvaka** [comment]
@kimwnasptd , any progress on this? Thank you.

**Comment by kimwnasptd** [comment]
@lalithvaka my only objection is about removing the inputs since the delete button looks too spaced out in the screenshots.
I'd suggest we keep the inputs as is, they are disabled either way so users wouldn't be able to modify them, and lets have a separate PR for hiding them.

the code is LGTM also from me

**Comment by lalithvaka** [comment]
@kimwnasptd , I have tested this feature with our end user community. They all liked the dropdown but they got confused on the "Size" and "Mode" fields. They were all assuming that any existing PVC they pick, they are expecting the fields disabled are reflection of what the actual size and mode of the selected PVC which is not the case. Hence to remove the confusion, I got rid of the fields on the selection of Existing PVCs. I totally agree with you that the spacing of the delete button needs some adjustment. 

Please let me know if you can approve or merge this?

**Comment by avdaredevil** [comment]
Moving ahead on this in favor of getting it in 1.1
/lgtm
/approve

Thanks for making this change @lalithvaka!

/cc @jlewi

**Comment by jlewi** [comment]
@lalithvaka Looks like there are test failures

**Comment by lalithvaka** [comment]
@jlewi , @kimwnasptd , @avdaredevil , pre submit test failures are fixed. Can you please lgtm / approve this PR? I would like to submit a cherry pick for this into 1.1. Thank you.

**Comment by thesuperzapper** [comment]
@lalithvaka there is an issue with this if you type the name of a notebook which has existed in past, the UI gets a bit confused, and changes to "existing" but leaves the "name" field as text field, rather than a dropdown.

**Comment by lalithvaka** [comment]
@thesuperzapper That was the behavior before I submitted the PR to show the drop down list for the existing PVCs. I thought about how to address that but couldn't come up with a better option with out totally revamping the existing design.  So I left it as it just shows the user that name is an existing PVC. I am not a UX/UI expert, if you think of anything better, please go ahead and submit an additional PR.

**Pull request merged**